src/rules/Username-attribute.js

```diff
@@ -0,0 +1,26 @@
+/**
+ * @title Add Username to AppMetadata 
+ * @overview Adds metadata on when an user first signs up or logs in. 
+ * @gallery true
+ * @category enrich profile
+ *
+ * This rule will add one attribute to the user's metadata object on when they log in or sign up
+ *
+ * This is useful for cases where you want to add the username to an email using liquid syntax.
+ */
+ 
+ function addPersistenceAttribute(user, context, callback) {
+    user.app_metadata = user.app_metadata || {};
+    // short-circuit if the user signed up already
+    if (user.app_metadata.username) return callback(null, user, context);
+    // first time login/signup
+    user.app_metadata.username = user.app_metadata.username || user.username;
+    context.idToken['https://example.com/favorite_color'] = user.app_metadata.username;
+    auth0.users.updateAppMetadata(user.user_id, user.app_metadata)
+      .then(function(){
+          callback(null, user, context);
+      })
+      .catch(function(err){
+          callback(err);
+      });
+  }
```

Comment:
Typically the rules are called something like "set-username-attribute" (all lowercase). Mind rename this file to match that format?

Reply:
I will fix and repushh

Reply:
@seejee Fixed